spec/xclarity_client_discovery_spec.rb

```diff
@@ -0,0 +1,37 @@
+require 'spec_helper'
+
+describe XClarityClient do
+
+  before :all do
+    WebMock.allow_net_connect! #-- Uncomment this line if you're testing with a external mock.
+
+    conf = XClarityClient::Configuration.new(
+    :username => ENV['LXCA_USERNAME'],
+    :password => ENV['LXCA_PASSWORD'],
+    :host     => ENV['LXCA_HOST'],
+    :port     => ENV['LXCA_PORT'],
+    :auth_type => ENV['LXCA_AUTH_TYPE'],
+    :verify_ssl => ENV['LXCA_VERIFY_SSL']
+    )
+
+    @client = XClarityClient::Client.new(conf)
+
+    @host = ENV['LXCA_HOST']
+  end
+
+  before :each do
+    @includeAttributes = %w(lastUpdateElapsedTime discoveryInProgress)
+    @excludeAttributes = %w(lastUpdateElapsedTime discoveryInProgress)
+  end
+
+  it 'has a version number' do
+    expect(XClarityClient::VERSION).not_to be nil
+  end
+
+  describe 'GET /discovery' do
+    it 'should respond with an array' do
+      expect(@client.discover_devices_by_slp).not_to be_empty
```

Comment:
Hi @ManasaRaoK, in the test below, I think that you are not testing if the result is an array
```
 it 'should respond with an array' do
￼   expect(@client.discover_devices_by_slp).not_to be_empty
end
```

I prefer that you use in instead
```
    it 'should respond with an array' do
      expect(@client.discover_devices_by_slp.class).to eq(Array)
    end
```

Reply:
Thanks for catching this. Addressed now.